### ISSUE TITLE
`redundant_clone` could also fire when the containing object will be dropped

### ISSUE THREAD

**Issue by sourcefrog** [rust-lang/rust-clippy#15858]
### What it does

Consider this code from [Zed](https://github.com/zed-industries/zed/blob/5698636c92d015738843ad1ce982b887a236110a/crates/rope/src/rope.rs#L121-L123):

```rust
    pub fn append(&mut self, rope: Rope) {
        // ....
        self.chunks.append(rope.chunks.clone(), ());
        self.check_invariants();
    }
```

`rope` is passed by value and dropped at the end of the function. I think we can eliminate the `clone` and just consume `rope.chunks`. 

Clippy doesn't warn about this even with `clippy::redundant_clones` turned on, I think because it's a more complex situation of consuming one field from a struct.
